libpod/options.go

```diff
@@ -2093,324 +2078,87 @@ func WithPodCgroup() PodCreateOption {
 }
 
 // WithInfraContainer tells the pod to create a pause container
-func WithInfraContainer() PodCreateOption {
+func WithInfraContainer(infra *specgen.SpecGenerator) PodCreateOption {
```

Comment:
This should not accept a specgen - take the *results* of the specgen (OCI spec and array of CtrCreateOption)

Reply:
Looking at it now, is this function even necessary? I can probably just pass nothing and simply have `pod.config.HasInfra = true` since the `isInfra` in the spec is already true

Reply:
You need some way to get the infra container config and options passed in, no?

Reply:
Fixed, just set `pod.config.HasInfra = true` no longer accepts  a spec